weld/vectorizer.rs

```diff
@@ -180,30 +199,27 @@ fn vectorizable(for_loop: &Expr<Type>) -> WeldResult<HashSet<Symbol>> {
                     body.traverse(&mut |f| {
                         match f.kind {
                             Literal(_) => {},
+
                             Ident(ref name) => {
                                 if f.ty == params[1].ty && *name == params[1].name {
                                     // Used an index expression in the loop body.
                                     passed = false;
                                 }
                             },
+
                             BinOp{ .. } => {},
+
                             Let{ ref name, .. } => {
                                 defined_in_loop.insert(name.clone()); 
                             },
-                            // GetField is allowed if the only fields we fetch are the ones from
-                            // the argument (in case the input was Zipped). At this point, it is
-                            // already guaranteed that each input vector is a scalar.
-                            GetField { ref expr, .. } => {
-                                let mut getfield_passed = false;
-                                let ref elem_param = params[2];
-                                if let Ident(ref name) = expr.kind {
-                                    if elem_param.name == *name {
-                                       getfield_passed = true; 
-                                    }
-                                }
-                                passed = getfield_passed;
-                            }
-                            Merge{ .. } => {},
```

Comment:
This should be okay...it works with nested structures and everything right? So basically, the behavior is to broadcast each literal in the struct?